server/webapp/WEB-INF/rails.new/app/controllers/analytics_controller.rb

```diff
@@ -44,7 +44,10 @@ def pipeline
   end
 
   private
+
   def render_plugin_error e
-    render :text => "Error generating analytics from plugin - #{params[:plugin_id]}", status: 500
```

Comment:
I am just wondering if we should retain the generic error message over using the exception message. User needs to know that something went wrong, not sure if the specific message from the exception would be useful to them. The logs should provide appropriate information for the admins to take action.

What do you think?

Reply:
I think it should be both:

1. Case 1: The plugin itself fails to respond properly. GoCD should show a meaningful message.

2. Case 2: The plugin understands that the settings are not provided. It should return a proper view path to an error page, with more details. Using the mechanism we use for displaying graphs.

Reply:
Per your comments @maheshp and @arvindsv I opted to keep a generic error message for the user (which would cover case 1), but go ahead and log the exception message for admins. 

As far as case 2, since we are using the mechanism provided for displaying graphs no changes to core were required. The changes have been made in the plugin.